discovery/watcher.go

```diff
@@ -411,17 +461,17 @@ func (w *Watcher) watchStream() (*addrSet, error) {
 		// This blocks until there is a change from the server.
 		resp, err := serverStream.Recv()
 		if err != nil {
-			w.log.Error("failed to parse server address from watch stream", "error", err)
+			w.log.Error("unable to receive from server watch stream", "error", err)
```

Comment:
I think this will log an error when `ctxForSwitch` is canceled to trigger the switch, which feels like it could cause some confusion for users as it's expected (not really an error case). Should we special-case it here?

```suggestion
		if !errors.Is(err, context.Canceled) {
			w.log.Error("unable to receive from server watch stream", "error", err)
		}
```